src/Command/GuidesCommand.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace PhpDocumentGenerator\Command;
+
+use PhpDocumentGenerator\Services\ConfigurationHandler;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\ArrayInput;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Style\SymfonyStyle;
+use Symfony\Component\Finder\Finder;
+
+final class GuidesCommand extends Command
+{
+    public function __construct(
+        private readonly ConfigurationHandler $configuration,
+        private readonly string $defaultTemplate
+    ) {
+        parent::__construct(name: 'guides');
+    }
+
+    protected function configure(): void
+    {
+        $this
+            ->setDescription('Creates Guides based on PHP code')
+            ->addOption(
+                name: 'output',
+                mode: InputArgument::REQUIRED,
+                description: 'The path where the guides will be printed'
+            )
+            ->addOption(
+                name: 'template',
+                mode: InputOption::VALUE_REQUIRED,
+                description: 'The path to the template file to use to generate each guide',
+                default: $this->defaultTemplate
+            );
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output): int
+    {
+        $style = new SymfonyStyle($input, $output);
+
+        $guideDir = $this->configuration->get('guides.src');
```

Comment:
can there be an option for that? We take option first, then configuration right?

Reply:
sure, on it !

Reply:
Currently, ReferencesCommand does not accept an input directory, it's based on the configuration (AFAIK)

Reply:
This can also be done if needed ?